### PR TITLE
Pttp 6320/dhcp script exclusions

### PR DESCRIPTION
**Pull request by wanieldilson**
# What

✨ Adding a feature to the DHCP script to output the exclusion information from a legacy DHCP export.

# Why

To make the pre go-live data checks easier

# Screenshots

image

# Notes

### CODE REVIEW

File: scripts/config-parser/lib/dhcp_config_parser.rb [hunk 1]
```diff
@@ -27,6 +33,22 @@ def self.export_file_exists?
     File.exist?(LEGACY_CONFIG_FILEPATH)
   end
 
+  def self.get_legacy_exclusions(export, subnet_list)
+    exclusion_fields = ["type", "start-ip", "end-ip"]
+    legacy_exclusions = []
+
+    subnet_list.each do |subnet|
+      exclusion_regex = /excluderange.#{subnet.chop}\d{1,3}.#{subnet.chop}\d{1,3}/
```
Comment (richrace) [inline scripts/config-parser/lib/dhcp_config_parser.rb:41]: Should the `.` in the regex be <any character> or <dot>. I.e. you want `\.` ?

I gues the same applies to line 70 now thinking about it.

**Comment by Gary-H9** [comment]
LGTM

**Comment by Gary-H9** [comment]
I wonder if its possible to add a tab between elements of the `exclusion_data` so that `end_ip` is aligned vertically at the end. Really this is a nice to have more than anything else.

**Comment by Gary-H9** [comment]
If `shared_network_id = "FITS_####"` is effectively a site ID then this could be used in line 18 of dhcp_config_parser.rb "This site ( #{shared_network_id}). If I've not understood correctly and it isn't then please ignore this comment.

File: scripts/config-parser/lib/dhcp_config_parser.rb (hunk 1 above)
Comment (wanieldilson) [reply scripts/config-parser/lib/dhcp_config_parser.rb:41]: It's actually just for the space between exclude range and the ip, but I guess I should probably use " " or [ \t]

Comment (wanieldilson) [reply scripts/config-parser/lib/dhcp_config_parser.rb:41]: But i have tested that regex on the full export of the entire DHCP server and didn't get any dodgy results so... 🤷

Comment (richrace) [reply scripts/config-parser/lib/dhcp_config_parser.rb:41]: Ohh that makes more sense.

**Review by richrace — APPROVED** [review]

**Pull request merged**